platforms/ios/lib/WysiwygComposer/Sources/HTMLParser/HTMLParser.swift

```diff
@@ -0,0 +1,127 @@
+            display: inline;
+            font-family: monospace;
+            white-space: pre;
+            -coretext-fontname: Menlo-Regular;
+            font-size: inherit;
+        }
+        h1,h2,h3 {
+            font-size: 1.2em;
+        }
+        """
+    }
+
+    private init() { }
+
+    // MARK: - Public
+
+    /// Parse given HTML to NSAttributedString with a standard style.
+    ///
+    /// - Parameters:
+    ///   - html: HTML to parse
+    ///   - encoding: string encoding to use
+    ///   - parserStyle: style to apply for HTML parsing
+    /// - Returns: an attributed string representation of the HTML content
+    public static func parse(html: String,
+                             encoding: String.Encoding = .utf16,
+                             parserStyle: HTMLParserStyle = .standard) throws -> NSAttributedString {
+        style = parserStyle
+
+        guard !html.isEmpty else {
+            return NSAttributedString(string: "")
+        }
+
+        // Fixes additionnal unrequired "\n" inserted by DTCoreText
+        var html = html
+        html = "<span>" + html + "</span>"
+        guard let data = html.data(using: encoding) else {
+            throw BuildHtmlAttributedError.dataError(encoding: encoding)
+        }
+
+        let defaultFont = UIFont.preferredFont(forTextStyle: .body)
+
+        let parsingOptions: [String: Any] = [
+            DTUseiOS6Attributes: true,
+            DTDefaultFontDescriptor: defaultFont.fontDescriptor,
+            DTDefaultStyleSheet: DTCSSStylesheet(styleBlock: defaultCSS) as Any,
+        ]
+
+        guard let builder = DTHTMLAttributedStringBuilder(html: data, options: parsingOptions, documentAttributes: nil) else {
+            throw BuildHtmlAttributedError.dataError(encoding: encoding)
+        }
+
+        guard let attributedString = builder.generatedAttributedString() else {
+            throw BuildHtmlAttributedError.dataError(encoding: encoding)
+        }
+
+        let mutableAttributedString = NSMutableAttributedString(attributedString: attributedString)
+
+        mutableAttributedString.addAttributes(
+            [.foregroundColor: style.textColor], range: NSRange(location: 0, length: mutableAttributedString.length)
+        )
+
+        // This fixes an iOS bug where if some text is typed after a link, and then a whitespace is added the link color is overridden.
+        mutableAttributedString.enumerateAttribute(
+            .link,
+            in: NSRange(location: 0, length: mutableAttributedString.length)
+        ) { value, range, _ in
+            if value != nil {
+                mutableAttributedString.removeAttribute(.underlineStyle, range: range)
+                mutableAttributedString.removeAttribute(.underlineColor, range: range)
+                mutableAttributedString.addAttributes([.foregroundColor: style.linkColor], range: range)
+            }
+        }
+
+        mutableAttributedString.applyBackgroundStyles()
```

Comment:
not a fan of this because... it uses the style of the HTMLParser behind curtains in its own implementation, I feel this function should have the style passed directly, and become `applyBackgroundStyles(style: HTMLParserStyle)`.